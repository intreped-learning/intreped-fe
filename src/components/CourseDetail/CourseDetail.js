import React from 'react';
import './CourseDetail.scss';
import { useSelector } from 'react-redux';
import YouTube from 'react-youtube';

const CourseDetail = ({ id }) => {
  const { courses } = useSelector(state => state);
  const currentCourse = courses.find(course => course.id === parseInt(id));
  const videoId = currentCourse.url.split('?v=')[1];
  const opts = {
    height: '390',
    width: '640',
    playerVars: { // https://developers.google.com/youtube/player_parameters
      autoplay: 0
    }
  };

  return (
    
    <div className="course-detail">
      <YouTube
        videoId={videoId}
        opts={opts}
      />
        <div className="course-info">
          <h3 className="category">{currentCourse.category}</h3>
          <h3 className="title">{currentCourse.title}</h3>
          <p className="description">{currentCourse.description}</p>
        </div>
      </div>
  );
};

export default CourseDetail;