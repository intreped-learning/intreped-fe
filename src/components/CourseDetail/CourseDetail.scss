.course-detail {
  width: 80%;
  margin: auto;
  margin-top: 80px;
}