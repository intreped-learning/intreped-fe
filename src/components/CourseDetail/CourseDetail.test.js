import React from 'react';
import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store'
import thunk from 'redux-thunk';
import { BrowserRouter as Router} from 'react-router-dom';
import CourseDetail from './CourseDetail';

const mockStore = configureMockStore([thunk]);
const store = mockStore({
  teacher: {
    username: '',
    id: null,
    teacher_courses: [
      {
        "model": "intreped.course",
        "id": 1,
        "fields": {
          "title": "Kagan Cooperative Learning: Inside-Outside Circle & Talking Chips",
          "description": "Kagan structures are techniques and activities teachers can use in their classrooms to keep students engaged, active, and have fun while they are learning",
          "url": "https://www.youtube.com/watch?v=B-hTXzUTxf8",
          "thumbnail": "https://i.ytimg.com/vi/hTXzUTxf8/mqdefault.jpg",
          "category": "Engagement Strategies",
          "duration": "11m30s"
        }
      },
      {
        "model": "intreped.course",
        "id": 2,
        "fields": {
          "title": "Think Pair Share Explained",
          "description": "This video provides a succinct explanation for students about how to participate in THINK PAIR SHARE activities.",
          "url": "https://www.youtube.com/watch?v=wW87rihT38I",
          "thumbnail": "https://i.ytimg.com/vi/wW87rihT38I/mqdefault.jpg",
          "category": "Engagement Strategies",
          "duration": "1m35s"
        }
      }
    ]
  },
  courses: [
    {
      "model": "intreped.course",
      "id": 1,
      "fields": {
        "title": "Kagan Cooperative Learning: Inside-Outside Circle & Talking Chips",
        "description": "Kagan structures are techniques and activities teachers can use in their classrooms to keep students engaged, active, and have fun while they are learning",
        "url": "https://www.youtube.com/watch?v=B-hTXzUTxf8",
        "thumbnail": "https://i.ytimg.com/vi/hTXzUTxf8/mqdefault.jpg",
        "category": "Engagement Strategies",
        "duration": "11m30s"
      }
    },
    {
      "model": "intreped.course",
      "id": 2,
      "fields": {
        "title": "Think Pair Share Explained",
        "description": "This video provides a succinct explanation for students about how to participate in THINK PAIR SHARE activities.",
        "url": "https://www.youtube.com/watch?v=wW87rihT38I",
        "thumbnail": "https://i.ytimg.com/vi/wW87rihT38I/mqdefault.jpg",
        "category": "Engagement Strategies",
        "duration": "1m35s"
      }
    }
  ],
  modalOpen: true
});
const getWrapper = () => mount(
  <Router>
    <Provider store={store}>
      <CourseDetail id={1}/>
    </Provider>
  </Router>
);
const wrapper = getWrapper();

describe('CourseDetail', () => {
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});