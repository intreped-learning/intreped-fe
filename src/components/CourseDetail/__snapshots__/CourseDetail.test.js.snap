// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseDetail should match snapshot 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "clearActions": [Function],
          "dispatch": [Function],
          "getActions": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
        }
      }
    >
      <CourseDetail
        id={1}
      >
        <div
          className="course-detail"
        >
          <YouTube
            className={null}
            containerClassName=""
            id={null}
            onEnd={[Function]}
            onError={[Function]}
            onPause={[Function]}
            onPlay={[Function]}
            onPlaybackQualityChange={[Function]}
            onPlaybackRateChange={[Function]}
            onReady={[Function]}
            onStateChange={[Function]}
            opts={
              Object {
                "height": "390",
                "playerVars": Object {
                  "autoplay": 1,
                  "controls": 0,
                  "disablekb": 1,
                },
                "width": "640",
              }
            }
            videoId="B-hTXzUTxf8"
          >
            <div
              className=""
            >
              <div
                className={null}
                id={null}
              />
            </div>
          </YouTube>
          <div
            className="course-info"
          >
            <h3
              className="category"
            />
            <h3
              className="title"
            />
            <p
              className="description"
            />
          </div>
        </div>
      </CourseDetail>
    </Provider>
  </Router>
</BrowserRouter>
`;
